### ISSUE TITLE
Exported script variables not showing up in inspector

### ISSUE THREAD

**Issue by Szesan** [godotengine/godot#50198]
### Godot version

3.3.2 mono

### System information

Windows 10

### Issue description

Adding a new variable with the **[Export]** keyword doesn't show up in the inspector. It was strange because I already had some other exported fields which showed up just fine.

After lot's of poking around, I realized that the inspector won't refresh until you rebuild the whole project. Which is fine, but there is no mention of this in the documentation:
https://docs.godotengine.org/en/stable/getting_started/scripting/gdscript/gdscript_exports.html

**Suggestion:** note this crucial detail in the aforementioned documentation. 

### Steps to reproduce

- Create a new project
- Add a scene
- Add a script to the scene
- Add a field with the [Export] attribute
- Save the script
- Open the scene in inspector

Following these steps the script variable won't show up.

**Comment by KoBeWi** [comment]
You need to recompile the script for the variables to appear.

**Comment by Torguen** [comment]
That's right, click on the "Build" button in the upper right corner of the editor.

**Comment by Calinou** [comment]
> After lot's of poking around, I realized that the inspector won't refresh until you rebuild the whole project. Which is fine, but there is no mention of this in the documentation:
docs.godotengine.org/en/stable/getting_started/scripting/gdscript/gdscript_exports.html

That documentation is about GDScript, not C#.

**Comment by SlitheringMule** [comment]
The editor has to recheck the script for changes, so it is logical that it does so after you save the script (effectively confirming that you want those changes). If you don't use an interpretted language I presume buliding is necesary for these changes 😄
